security/auth.go is out of date with goa v3

Noticed this while trying to reproduce an example to look at https://github.com/goadesign/goa/issues/2116

The security example builds but has a runtime panic.

$ cd security/
$ GO111MODULE=on go build ./cmd/multi_auth
$ ./multi_auth 
panic: interface conversion: *multiauth.securedServiceSvc is not securedservice.Auther: missing method APIKeyAuth

goroutine 1 [running]:
goa.design/examples/security/gen/secured_service.NewEndpoints(0xa80200, 0xc0000ac568, 0x0)
        /home/eric/go/src/goa.design/examples/security/gen/secured_service/endpoints.go:30 +0x59
main.main()
        /home/eric/go/src/goa.design/examples/security/cmd/multi_auth/main.go:53 +0x32a

No time to fix it tonight, I'll find a few minutes to create a PR tomorrow.

--- a/security/auth.go
+++ b/security/auth.go
@@ -24,9 +24,10 @@ var (
 	Key = []byte("secret")
 )
 
-// SecuredServiceBasicAuth implements the authorization logic for service
-// "secured_service" for the "basic" security scheme.
-func SecuredServiceBasicAuth(ctx context.Context, user, pass string, s *security.BasicScheme) (context.Context, error) {
+
+// BasicAuth implements the authorization logic for service "secured_service"
+// for the "basic" security scheme.
+func (s *securedServiceSvc) BasicAuth(ctx context.Context, user, pass string, scheme *security.BasicScheme) (context.Context, error) {
 	if user != "goa" {
 		return ctx, ErrUnauthorized
 	}
@@ -36,9 +37,9 @@ func SecuredServiceBasicAuth(ctx context.Context, user, pass string, s *security
 	return ctx, nil
 }
 
-// SecuredServiceJWTAuth implements the authorization logic for service
-// "secured_service" for the "jwt" security scheme.
-func SecuredServiceJWTAuth(ctx context.Context, token string, s *security.JWTScheme) (context.Context, error) {
+// JWTAuth implements the authorization logic for service "secured_service" for
+// the "jwt" security scheme.
+func (s *securedServiceSvc) JWTAuth(ctx context.Context, token string, scheme *security.JWTScheme) (context.Context, error) {
 	claims := make(jwt.MapClaims)
 
 	// authorize request
@@ -60,24 +61,24 @@ func SecuredServiceJWTAuth(ctx context.Context, token string, s *security.JWTSch
 	for _, scp := range scopes {
 		scopesInToken = append(scopesInToken, scp.(string))
 	}
-	if err := s.Validate(scopesInToken); err != nil {
+	if err := scheme.Validate(scopesInToken); err != nil {
 		return ctx, securedservice.InvalidScopes(err.Error())
 	}
 	return ctx, nil
 }
 
-// SecuredServiceAPIKeyAuth implements the authorization logic for service
-// "secured_service" for the "api_key" security scheme.
-func SecuredServiceAPIKeyAuth(ctx context.Context, key string, s *security.APIKeyScheme) (context.Context, error) {
+// APIKeyAuth implements the authorization logic for service "secured_service"
+// for the "api_key" security scheme.
+func (s *securedServiceSvc) APIKeyAuth(ctx context.Context, key string, scheme *security.APIKeyScheme) (context.Context, error) {
 	if key != "my_awesome_api_key" {
 		return ctx, ErrUnauthorized
 	}
 	return ctx, nil
 }
 
-// SecuredServiceOAuth2Auth implements the authorization logic for service
-// "secured_service" for the "oauth2" security scheme.
-func SecuredServiceOAuth2Auth(ctx context.Context, token string, s *security.OAuth2Scheme) (context.Context, error) {
+// OAuth2Auth implements the authorization logic for service "secured_service"
+// for the "oauth2" security scheme.
+func (s *securedServiceSvc) OAuth2Auth(ctx context.Context, token string, scheme *security.OAuth2Scheme) (context.Context, error) {
 	claims := make(jwt.MapClaims)
 
 	// authorize request
@@ -99,7 +100,7 @@ func SecuredServiceOAuth2Auth(ctx context.Context, token string, s *security.OAu
 	for _, scp := range scopes {
 		scopesInToken = append(scopesInToken, scp.(string))
 	}
-	if err := s.Validate(scopesInToken); err != nil {
+	if err := scheme.Validate(scopesInToken); err != nil {
 		return ctx, securedservice.InvalidScopes(err.Error())
 	}
 	return ctx, nil

--- a/security/auth.go
+++ b/security/auth.go
@@ -24,7 +24,6 @@ var (
 	Key = []byte("secret")
 )
 
-
 // BasicAuth implements the authorization logic for service "secured_service"
 // for the "basic" security scheme.
 func (s *securedServiceSvc) BasicAuth(ctx context.Context, user, pass string, scheme *security.BasicScheme) (context.Context, error) {

